salt/modules/boto3_route53.py

```diff
@@ -1000,3 +1003,448 @@ def change_resource_record_sets(HostedZoneId=None, Name=None,
+    CLI Example:
+    .. code-block:: bash
+        salt myminion boto3_route53.get_health_check_version_by_name ANAME \
+                profile='{"region": "us-east-1", "keyid": "A12345678AB", "key": "xblahblahblah"}'
+    '''
+    data = find_health_check(None, Name, region=region, key=key, keyid=keyid, profile=profile)
+    return data['HealthCheckVersion'] if data else None
+
+def create_health_check(Name, CallerReference=None, IPAddress=None,
+                        Port=None, Type=None, ResourcePath=None, FullyQualifiedDomainName=None, SearchString=None,
+                        RequestInterval=None, FailureThreshold=None, MeasureLatency=None, Inverted=None, Disabled=None,
+                        HealthThreshold=None, ChildHealthChecks=None, EnableSNI=None, Regions=None, AlarmIdentifier=None,
+                        InsufficientDataHealthStatus=None, region=None, key=None, keyid=None, profile=None, tries=20):
+    '''
+    Creates a new health check with given properties.
+
+    Name
+        The name to assign to newly created health check.
+    region
+        Region to connect to.
+    key
+        Secret key to be used.
+    keyid
+        Access key to be used.
+    profile
+        Dict, or pillar key pointing to a dict, containing AWS region/key/keyid.
+    CLI Example:
+    .. code-block:: bash
+        salt myminion boto3_route53.create_health_check ANAME ... \
+                profile='{"region": "us-east-1", "keyid": "A12345678AB", "key": "xblahblahblah"}'
+    '''
+    if not Name:
+        raise SaltInvocationError('Name of healthcheck to update missing.')
+
+    keys = [
+        'IPAddress', 'Port', 'Type', 'ResourcePath', 'FullyQualifiedDomainName', 'SearchString',
+        'RequestInterval', 'FailureThreshold', 'MeasureLatency', 'Inverted', 'Disabled', 'HealthThreshold',
+        'EnableSNI', 'InsufficientDataHealthStatus', 'Regions', 'AlarmIdentifier', 'ChildHealthChecks'
+    ]
+    params = locals()
+    result = None
+
+    conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
+    CallerReference = CallerReference if CallerReference else str(uuid.uuid4())  # future lint: disable=blacklisted-function
+    HealthCheckConfig = {}
+
+    for key in [x for x in keys if params.get(x) is not None]:
+        HealthCheckConfig[key] = params.get(key)
+
+    args = {'CallerReference': CallerReference, 'HealthCheckConfig': HealthCheckConfig}
+
+    while tries:
+        try:
+            result = conn.create_health_check(**args).get('HealthCheck')
+            conn.change_tags_for_resource(
+                ResourceType='healthcheck',
+                ResourceId=result['Id'],
+                AddTags=[{ 'Key': 'Name', 'Value': Name }],
+                )
+            return {'result': result }
+        except ClientError as e:
+            if tries and e.response.get('Error', {}).get('Code') == 'Throttling':
+                log.debug('Throttled by AWS API.')
+                time.sleep(3)
```

Comment:
Should we do an exponential/increasing backoff here? Or is 3 seconds the right way to go?

Reply:
Dunno, this is taken from another boto3 module. I am not keep on adding sleeps to code, but given it was under an exception handler, and I was not able to repro the actual issue, I simply relied on what was already there.

Reply:
C'mon, that's easy, all you need to do is [create 200(and one?) active health checks (per account??)](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/DNSLimitations.html#limits-api-entities-health-checks) :trollface:

I guess I'm OK with this. I'd prefer something else, like you, but hopefully it shouldn't happen very often. Maybe just add a comment to the code suggesting that we could update the way we do the backoffs to actually check for how long we need to wait, or just report the failure or something.